meta/user/user.ts

```diff
@@ -15,6 +15,6 @@ export interface User {
   position?: string
   profilePictureFile?: string
   profilePictureFilename?: string
-  roles: Array<UserRole<RoleName>>
+  roles: Array<UserRole<RoleName> | Collaborator>
```

Comment:
isn't Collaborator a UserRole ? you can remove this

Reply:
it's a specific type of userrole without void as props. Without this OR, the application won't work. This happened because the query returns an array of users and from the type User it's impossible to specify the type of the roles which is UserRole<RoleName, void> by default.

Reply:
I am running in similar issue(s) where I have to specifically define roles to be Collaborator in some parts, otherwise I will get typescript errors as mentioned above

I have changed the void to undefined though, as void should we used as function return type

Reply:
you're right, with undefined we can cast the props field without typescript errors